### PR TITLE
[CAT-36, CAT-37, CAT-38] Badges

### PR DESCRIPTION
**Pull request by JonathanAaron3005**

### CODE REVIEW

File: Mini2_test/Model/Food.swift
Comment (theodevoid) [inline Mini2_test/Model/Food.swift:1]: Should be deleted

File: Mini2_test/View/Presenter/BadgesView.swift [hunk 1]
```diff
@@ -110,19 +107,49 @@ struct BadgesView: View {
...
-                        .padding(.top, 20)
+                    Button{
+                        guard let selectedBadge = selectedBadge else
+                        { return }
+                        
+                        if selectedBadge.isUnlocked{
+                            assignBadgeToCat(selectedBadge)
+                        } else {
+                            buyBadge(badge: selectedBadge)
+                        }
```
Comment (theodevoid) [inline Mini2_test/View/Presenter/BadgesView.swift]: separate to a function, more readable

File: Mini2_test/View/Presenter/BadgesView.swift [hunk 2]
```diff
@@ -110,19 +107,49 @@ struct BadgesView: View {
...
+    }
+    
+    private func buyBadge(badge: Badge){
+        if user[0].coin >= badge.price {
+            badge.isUnlocked = true
+            user[0].coin -= badge.price
+        } else {
+            alertMessage = "You need \(badge.price - user[0].coin) more coins to buy this badge."
+            showAlert = true
+        }
```
Comment (theodevoid) [inline Mini2_test/View/Presenter/BadgesView.swift]: separate logic to ViewModel

File: Mini2_test/Model/FoodInfo.swift [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+//
+
+import Foundation
+
+struct FoodInfo: Codable {
+    let food_id: String
+    let fv_grade: String
+    let g_per_serving: Double
+    let display_name: String
+    let nutrition: Nutrition
```
Comment (theodevoid) [inline Mini2_test/Model/FoodInfo.swift:15]: use camel case

File: Mini2_test/Model/Nutrition.swift [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    let vitamin_b3_100g: Double?
+    let vitamin_b5_100g: Double?
+    let vitamin_b6_100g: Double?
+    let vitamin_b9_100g: Double?
+    let vitamin_c_100g: Double?
+    let vitamin_d_100g: Double?
+    let vitamin_e_100g: Double?
+    let vitamin_k1_100g: Double?
+    let water_100g: Double?
+    let zinc_100g: Double?
```
Comment (theodevoid) [inline Mini2_test/Model/Nutrition.swift:45]: use camel case here too

**Review by khoirunnisarizky — APPROVED** [review]
looking for the code review notes

File: Mini2_test/View/Presenter/HomeView.swift [hunk 5]
```diff
@@ -34,6 +34,8 @@ struct HomeView: View {
                         .frame(width: 200)
                         .offset(x: 20)
                     
+                    //foreach badge, offset => badge.x badge.y
```
Comment (theodevoid) [inline Mini2_test/View/Presenter/HomeView.swift]: comment ketinggalan

File: Mini2_test/View/Presenter/HomeView.swift [hunk 6]
```diff
@@ -89,97 +91,102 @@ struct HomeView: View {
...
-    //                                                  carbohydrates: 150,
-    //                                                  fat: 45)
-    //        )
-    //    }
+            do {
+                try modelContext.save()
+            } catch {
+                print("Error saving context after deletion: \(error.localizedDescription)")
+            }
+        }
```
Comment (theodevoid) [inline Mini2_test/View/Presenter/HomeView.swift]: separate to a view model pls

**Review by theodevoid — CHANGES_REQUESTED** [review]
few notes, gas terus!

File: Mini2_test/View/Presenter/BadgesView.swift (hunk 2 above)
Comment (JonathanAaron3005) [reply Mini2_test/View/Presenter/BadgesView.swift]: keknya ini nanti aja, uda tny ka khoi masi gabisa. Nanti kalo sempet baru ubah lagi ke MVVM

File: Mini2_test/View/Presenter/HomeView.swift (hunk 6 above)
Comment (JonathanAaron3005) [reply Mini2_test/View/Presenter/HomeView.swift]: nanti kalo sempet

**Review by theodevoid — APPROVED** [review]
LGTM 🚀

**Pull request merged**